test/unit/main_test.go

```diff
@@ -0,0 +1,567 @@
+			ExtraJvmOpts: strings.Join(extraJavaOptions, " "),
+		},
+		Replicas: operatorReplicas,
+	},
+}
+
+var infinispanEndpointEncryption = ispnv1.Infinispan{
+	TypeMeta: metav1.TypeMeta{
+		APIVersion: operatorAPIVersion,
+		Kind:       operatorKind,
+	},
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      operatorName,
+		Namespace: operatorNamespace,
+	},
+	Spec: ispnv1.InfinispanSpec{
+		Replicas: operatorReplicas,
+		Security: ispnv1.InfinispanSecurity{
+			EndpointEncryption: ispnv1.EndpointEncryption{
+				Type:           "secret",
+				CertSecretName: operatorTlsSecretName,
+			},
+		},
+	},
+}
+
+var infinispanWithCustomSecret = ispnv1.Infinispan{
+	TypeMeta: metav1.TypeMeta{
+		APIVersion: operatorAPIVersion,
+		Kind:       operatorKind,
+	},
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      operatorName,
+		Namespace: operatorNamespace,
+	},
+	Spec: ispnv1.InfinispanSpec{
+		Replicas: operatorReplicas,
+		Security: ispnv1.InfinispanSecurity{
+			EndpointSecretName: endpointCustomSecretName + "1",
+		},
+	},
+}
+
+var endpointEncryptionWithService = ispnv1.InfinispanSecurity{
+	EndpointEncryption: ispnv1.EndpointEncryption{
+		Type:            "service",
+		CertServiceName: operatorTlsServiceName,
+	},
+}
+
+var endpointEncryptionSecret = &corev1.Secret{
+	Type: corev1.SecretTypeOpaque,
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      operatorTlsSecretName,
+		Namespace: operatorNamespace,
+	},
+	StringData: map[string]string{
+		"alias":    "server-alias",
+		"password": "password-password",
+	},
+	Data: map[string][]byte{
+		"keystore.p12": []byte("KEYSTORE.P12"),
+		"alias":        []byte("server-alias"),
+		"password":     []byte("password-password"),
+	},
+}
+
+var endpointEncryptionSecretTlsData = map[string][]byte{
+	"tls.key": []byte("TLS.KEY"),
+	"tls.crt": []byte("TLS.CRT"),
+}
+
+var req = reconcile.Request{
+	NamespacedName: types.NamespacedName{
+		Name:      operatorName,
+		Namespace: operatorNamespace,
+	},
+}
+
+var basicRestMock = func(r *http.Request) (*http.Response, error) {
+	notFoundResp := &http.Response{
+		Status:        string(http.StatusNotFound),
+		StatusCode:    http.StatusNotFound,
+		Proto:         "HTTP/1.1",
+		ProtoMajor:    1,
+		ProtoMinor:    1,
+		Body:          ioutil.NopCloser(bytes.NewBufferString("")),
+		ContentLength: 0,
+		Request:       r,
+		Header:        make(http.Header, 0),
+	}
+
+	if r.Method == http.MethodGet && r.URL.Path == "/apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions/servicecas.operator.openshift.io" {
+		return notFoundResp, nil
+	}
+	return nil, nil
+}
+
+func TestBasicConfiguration(t *testing.T) {
+	infinispan := basicInfinispan
+	r := testDefaultConfiguration(t, infinispan, basicRestMock)
+	sset := tstutil.GetStatefulSet(r.GetClient(), types.NamespacedName{operatorNamespace, operatorName})
+
+	assert.Equal(t, corev1.URIScheme("HTTP"), sset.Spec.Template.Spec.Containers[0].LivenessProbe.HTTPGet.Scheme, "Liveness probe Scheme not defined as expected")
```

Comment:
Should this checks be part of `testDefaultConfiguration`?

Reply:
I will look into it. Thanks.

Reply:
@Crumby looks like it's a correct test, as testDefaultConfiguration uses for both URI schema's: HTTP and HTTPS